services/spar/src/Spar/Types.hs

```diff
@@ -23,7 +23,40 @@ import URI.ByteString
 import qualified Data.X509 as X509
 
 
-type IdP = IdPConfig TeamId
+-- | Info about the service provider that can be given to the identity
+-- provider to configure the service provider.
+data SPInfo = SPInfo
+  { _spiMetaURI  :: URI        -- ^ Unique for Wire
+  , _spiLoginURI :: URI        -- ^ Different for each identity provider
+  }
+  deriving (Eq, Show, Generic)
+
+makeLenses ''SPInfo
+deriveJSON deriveJSONOptions ''SPInfo
+
+-- | The identity provider type used in Spar.
+type IdP = IdPConfig IdPExtra
+
+-- | Extra information stored for each 'IdP'. The SAML library handles it
+-- but never inspects it (see the '_idpExtraInfo' field).
+data IdPExtra = IdPExtra
+  { _idpeTeam   :: TeamId
+  , _idpeSPInfo :: SPInfo
+  }
+  deriving (Eq, Show, Generic)
+
+makeLenses ''IdPExtra
+deriveJSON deriveJSONOptions ''IdPExtra
+
+-- | A list of 'IdP's, returned by some endpoints. Wrapped into an object to
+-- allow extensibility later on.
+data IdPList = IdPList
```

Comment:
I think that similarly to https://github.com/wireapp/wire-server/pull/402, we should make this an object